workflow/type.go

```diff
@@ -22,23 +22,32 @@ const (
 // Workflow describes a workflow of a service
 type Workflow struct {
 	Hash    hash.Hash `hash:"-" validate:"required"`
-	Trigger *Trigger  `hash:"name:1" validate:"required"`
-	Tasks   []*Task   `hash:"name:2" validate:"required"`
-	Key     string    `hash:"name:3" validate:"required"`
+	Key     string    `hash:"name:1" validate:"required"`
+	Trigger Trigger   `hash:"name:2" validate:"required"`
+	Nodes   []Node    `hash:"name:3" validate:"dive,required"`
```

Comment:
What do you think about moving the graph as a separate structure? In that case you can control and test all graph functions like searching path, walking order and more without Workflow and workflow will just use it like:

```go
type Workflow struct {
    // ....
    g wgraph
}
```

Reply:
for now, the workflow is pretty much only the graph but yes I agree that could be nice. I would still keep that as a future refactoring if the package gets bigger or we need to reuse the graph